### ISSUE TITLE
fup to : #3928 the current ansible-lint github workflow is using the latest checkout from main, indifferent of the version specified

### ISSUE THREAD

**Issue by zerwes** [ansible/ansible-lint#3932]
##### Summary

see https://github.com/ansible/ansible-lint/issues/3928#issuecomment-1842753891

even if I specify a tag/version in my github action, the ansible-lint workflow will run against main

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

github workflow

##### STEPS TO REPRODUCE
ansible-lint workflow:
```yml
uses: ansible/ansible-lint@v6.22.0
```
will run against main
```
Run ansible/ansible-lint@v6.22.0
  with:
    setup_python: true
Run wget --output-document=.git/ansible-lint-requirements.txt https://raw.githubusercontent.com/ansible/ansible-lint/main/.config/requirements-lock.txt
...
Building wheels for collected packages: ansible-lint
175
  Building wheel for ansible-lint (pyproject.toml): started
176
  Building wheel for ansible-lint (pyproject.toml): finished with status 'done'
177
  Created wheel for ansible-lint: filename=ansible_lint-6.22.2.dev8-py3-none-any.whl size=297703 
...
ansible-lint 6.22.2.dev8 using ansible-core:2.16.0 ansible-compat:4.1.10 ruamel-yaml:0.18.5 ruamel-yaml-clib:0.2.8
You are using a pre-release version of ansible-lint.
Run ansible-lint
...
```

**Comment by zerwes** [comment]
https://github.com/Rosa-Luxemburgstiftung-Berlin/ansible-opnsense/actions/runs/7114305453/job/19368083185

**Comment by zerwes** [comment]
while debuging in my fork:
the var `github.action_ref` seems empty

**Comment by zerwes** [comment]
https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability

> github.action_ref
> string
> For a step executing an action, this is the ref of the action being executed. For example, v2.
> Do not use in the run context. To make this context work with composite actions, reference it within the env context of the composite action.

**Comment by zerwes** [comment]
https://github.com/ansible/ansible-lint/issues/3928#issuecomment-1842875546
would be great if a fix for versions before 6.22.1 would be applied

**Comment by zerwes** [comment]
We migrated all our to 6.22.2 as no backward fix was in view.
So for my part this is not relevant anymore.
